tests/testthat/test-get_chol.R

```diff
@@ -0,0 +1,103 @@
+# h_get_chol ----
+
+test_that("h_get_chol works correctly", {
+  fit <- get_mmrm()
+  expect_identical(
+    fit$tmb_object$report()$cholesky_all,
+    h_get_chol(fit)
+  )
+  new_theta <- seq_len(length(fit$theta_est)) / length(fit$theta_est)
+  expect_identical(
+    fit$tmb_object$report(new_theta)$cholesky_all,
+    h_get_chol(fit, theta = new_theta, data = fev_data, weights = NULL)
+  )
+})
+
+test_that("h_get_chol works correctly to provide cholesky on all visits", {
+  fit <- get_mmrm()
+  new_theta <- seq_len(length(fit$theta_est)) / length(fit$theta_est)
+  chols <- expect_silent(
+    h_get_chol(fit, theta = new_theta, data = fev_data, weights = NULL, complete_case = FALSE)
+  )
+  lapply(
+    chols,
+    expect_matrix,
+    ncols = 4,
+    nrows = 4
+  )
+})
+
+test_that("h_get_chol errors on non correct theta", {
+  fit <- get_mmrm()
+  expect_error(
+    h_get_chol(fit, theta = 1),
+    "Must have length 10"
+  )
+})
+
+test_that("h_get_chol errors on non correct data", {
+  fit <- get_mmrm()
+  expect_error(
+    h_get_chol(fit, data = iris),
+    "Names must include the elements"
+  )
+})
+
+test_that("h_get_chol works with visit var as character subset of original fit", {
```

Comment:
can you add a test where we have additional subjects and also additional visits?
so e.g. original data only has subjects 1, 2, now we have 3 as well.
Also original data has only visits 1, 4, 5, for subject 1, now we look at visits 2, 3, 4, 5, 6.

Reply:
this is already covered in https://github.com/openpharma/mmrm/blob/15c7dae2e45946266a9fba5d66b8ccedeee6e8fe/tests/testthat/test-get_chol.R#L73 where the subject id are totally new (not part of the orginal PT1 things)

and also https://github.com/openpharma/mmrm/blob/15c7dae2e45946266a9fba5d66b8ccedeee6e8fe/tests/testthat/test-get_chol.R#L22 the visits provided is more than previously we have (all subjects has 4 visits)

and also https://github.com/openpharma/mmrm/blob/15c7dae2e45946266a9fba5d66b8ccedeee6e8fe/tests/testthat/test-get_chol.R#L65

additional visits will cause error

Reply:
hm ok I did not see that clearly from description. can you please add additional test with corresponding clear label? thanks